### PR TITLE
Update Kafka PDB to use live Pods instead of the CR Spec

### PR DESCRIPTION
**Pull request by alungu**
| Q               | A      |
| --------------- | ------ |
| Bug fix?        | yes |
| New feature?    | no |
| API breaks?     | no |
| Deprecations?   | no |
| Related tickets | TBD |
| License         | Apache 2.0 |

### What's in this PR?
The Kafka PDB logic is update to consider the total number of healthy Kafka brokers as being the number of Running Kafka pods (instead of the number of Kafka brokers specified in the Kafka CR)

### Why?
The previous implementation lead to a faulty PDB during scaling (both in and out) since adding or removing brokers requires some time.
Also, if there will be a broken broker, CruiseControl will evict that broker. As such, we should be able to allow disruptions into the clusters (within the set budget)

### Checklist

- [ X] Implementation tested
- [X ] Error handling code meets the [guideline](https://github.com/banzaicloud/developer-guide/blob/master/docs/coding-style/error-handling-guide.md)
- [ X] Logging code meets the guideline

### CODE REVIEW

**Review by amuraru — APPROVED** [review]

**Comment by amuraru** [comment]
worth filing this PR to upstream as well

File: pkg/resources/kafka/poddisruptionbudget.go [hunk 1]
```diff
@@ -62,6 +66,21 @@ func (r *Reconciler) podDisruptionBudget(log logr.Logger) (runtime.Object, error
...
+
+	// number of healthy brokers in the KafkaCluster
+	brokers := 0
+	for _, pod := range podList.Items {
+		if pod.Status.Phase == corev1.PodRunning {
+			brokers++
+		}
+	}
+
 	/*
```
Comment (amuraru) [inline pkg/resources/kafka/poddisruptionbudget.go:84]: this inline topic was meant to be at the top of the method body. let's keep it there? or move it as a godoc

Comment (alungu) [reply pkg/resources/kafka/poddisruptionbudget.go]: done!

**Review by amuraru — APPROVED** [review]
looks good - please also file this PR against the upstream repo as well.

**Review by ecojan — APPROVED** [review]
Had a look and busted my brain for scenarios, this solution seems to be the best one so far.

**Comment by ecojan** [comment]
Scenarios I went through:

We assume a budget of 1 pod down for all Scenarios

=============== Base scenarios ===============

1. Assume we are working with 3 brokers that are up and running
Operator targets 3 pods -> 3 - 1 = min available 2
PDB sees 3 available pods => 3 - 2 = 1 allowed disruption

2. Assume we are working with 3 brokers where only 2 are up and running, one is in a non-running state
Operator targets 3 pods -> 3 - 1 = min available 2
PDB sees 2 available pods => 2 - 2 = 0 allowed disruption

This case should have an alert if it stays this way for let's say 30minutes


=============== Normal operations scenarios ===============

3. Scale up scenario
Same scenario as with #2

4. Scale down scenario
Cluster goes into scenario #2 and moves to #1 after the pod dissapears

=============== Operations during maintenance ===============

5. Scale up during maintenance
Same as scenario #2 and #3

6. Scale down during maintenance
Same as scenario #4

The assumption is here that the operator that's draining retries after an X amount of time for the Cluster to exit scenarios similar to #2

=============== Operator Down during maintenance ===============

7. The operator is down and maintenance is being done
Will move from scenario #1 to #2 and stay this way until operator is recovered

The assumption is here that the operator that's draining retries after an X amount of time for the Cluster to exit scenarios similar to #2

**Pull request merged**